diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN mplayer-20090226.28734.orig/configure mplayer-20090226.28734/configure
--- mplayer-20090226.28734.orig/configure	2009-02-26 12:25:42.000000000 -0500
+++ mplayer-20090226.28734/configure	2009-03-02 15:10:12.000000000 -0500
@@ -1349,14 +1349,7 @@ if test -z "$_target" ; then
       case "`uname -m 2>&1`" in
       i[3-9]86*|x86|x86pc|k5|k6|k6_2|k6_3|k6-2|k6-3|pentium*|athlon*|i586_i686|i586-i686|BePC) host_arch=i386 ;;
       ia64) host_arch=ia64 ;;
-      x86_64|amd64)
-        if [ -n "$($_cc -dumpmachine | sed -n '/^x86_64-/p;/^amd64-/p')" -a \
-             -z "$(echo $CFLAGS $_cc | grep -- -m32)" ]; then
-          host_arch=x86_64
-        else
-          host_arch=i386
-        fi
-      ;;
+      x86_64|amd64) host_arch=i386 ;;
       macppc|ppc|ppc64) host_arch=ppc ;;
       alpha) host_arch=alpha ;;
       sparc) host_arch=sparc ;;
