diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN x264-snapshot-20081218-2245.orig/configure x264-snapshot-20081218-2245/configure
--- x264-snapshot-20081218-2245.orig/configure	2008-12-18 16:45:07.000000000 -0500
+++ x264-snapshot-20081218-2245/configure	2009-01-16 16:05:42.581069829 -0500
@@ -236,7 +236,7 @@ case $host_os in
 esac
 
 case $host_cpu in
-  i*86)
+  i*86|x86_64)
     ARCH="X86"
     ASFLAGS="-O2"
     if [ "$SYS" = MACOSX ]; then
@@ -247,16 +247,6 @@ case $host_cpu in
       ASFLAGS="$ASFLAGS -f elf"
     fi
     ;;
-  x86_64)
-    ARCH="X86_64"
-    if [ "$SYS" = MACOSX ];then
-      ASFLAGS="-f macho64 -m amd64 -DPIC -DPREFIX"
-      CFLAGS="$CFLAGS -arch x86_64"
-      LDFLAGS="$LDFLAGS -arch x86_64"
-    else
-      ASFLAGS="-f elf -m amd64"
-    fi
-    ;;
   powerpc|powerpc64)
     ARCH="PPC"
     if [ $SYS = MACOSX ]
