Index: toolchain.eclass
===================================================================
RCS file: /var/cvsroot/gentoo-x86/eclass/toolchain.eclass,v
retrieving revision 1.458
diff -u -p -r1.458 toolchain.eclass
--- toolchain.eclass	10 Apr 2011 18:21:20 -0000	1.458
+++ toolchain.eclass	11 Apr 2011 23:31:09 -0000
@@ -1,4 +1,4 @@
-# Copyright 1999-2008 Gentoo Foundation
+# Copyright 1999-2010 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo-x86/eclass/toolchain.eclass,v 1.458 2011/04/10 18:21:20 vapier Exp $
 #
@@ -1241,7 +1241,11 @@ gcc-compiler-configure() {
 		# Default arch for x86 is normally i386, lets give it a bump
 		# since glibc will do so based on CTARGET anyways
 		x86)
-			confgcc="${confgcc} --with-arch=${CTARGET%%-*}"
+			if ! is_multilib ; then
+				# Don't set default arch for multilib; it breaks 64bit compiles
+				# unless you give -march, which breaks the gcc build itself.
+				confgcc="${confgcc} --with-arch=${CTARGET%%-*}"
+			fi
 			;;
 		# Enable sjlj exceptions for backward compatibility on hppa
 		hppa)
@@ -1448,15 +1452,15 @@ gcc_do_configure() {
 	esac
 	[[ ${GCCMAJOR}.${GCCMINOR} < 3.4 ]] && confgcc="${confgcc} --disable-libunwind-exceptions"
 
-	# create a sparc*linux*-{gcc,g++} that can handle -m32 and -m64 (biarch)
-	if [[ ${CTARGET} == sparc*linux* ]] \
-		&& is_multilib \
-		&& ! is_crosscompile \
-		&& [[ ${GCCMAJOR}.${GCCMINOR} > 4.2 ]]
-	then
-		confgcc="${confgcc} --enable-targets=all"
+	if [[ ${CTARGET} == *linux* ]] && is_multilib && ! is_crosscompile &&
+		[[ ${GCCMAJOR}.${GCCMINOR} > 4.2 ]] ; then
+			case $(tc-arch) in
+				sparc|x86) confgcc="${confgcc} --enable-targets=all"
+				;;
+			esac
 	fi
 
+
 	tc_version_is_at_least 4.3 && set -- "$@" \
 		--with-bugurl=http://bugs.gentoo.org/ \
 		--with-pkgversion="${BRANDING_GCC_PKGVERSION}"
