Index: toolchain.eclass
===================================================================
RCS file: /var/cvsroot/gentoo-x86/eclass/toolchain.eclass,v
retrieving revision 1.399
diff -u -p -r1.399 toolchain.eclass
--- toolchain.eclass	24 May 2009 16:57:43 -0000	1.399
+++ toolchain.eclass	27 May 2009 14:30:03 -0000
@@ -1225,7 +1225,11 @@ gcc-compiler-configure() {
 		# Default arch for x86 is normally i386, lets give it a bump
 		# since glibc will do so based on CTARGET anyways
 		x86)
-			confgcc="${confgcc} --with-arch=${CTARGET%%-*}"
+			if ! is_multilib ; then
+				# Don't set default arch for multilib; it breaks 64bit compiles
+				# unless you give -march, which breaks the gcc build itself.
+				confgcc="${confgcc} --with-arch=${CTARGET%%-*}"
+			fi
 			;;
 		# Enable sjlj exceptions for backward compatibility on hppa
 		hppa)
@@ -1397,15 +1401,15 @@ gcc_do_configure() {
 	fi
 	[[ ${GCCMAJOR}.${GCCMINOR} < 3.4 ]] && confgcc="${confgcc} --disable-libunwind-exceptions"
 
-	# create a sparc*linux*-{gcc,g++} that can handle -m32 and -m64 (biarch)
-	if [[ ${CTARGET} == sparc*linux* ]] \
-		&& is_multilib \
-		&& ! is_crosscompile \
-		&& [[ ${GCCMAJOR}.${GCCMINOR} > 4.2 ]]
-	then
-		confgcc="${confgcc} --enable-targets=all"
+	if [[ ${CTARGET} == *linux* ]] && is_multilib && ! is_crosscompile &&
+		[[ ${GCCMAJOR}.${GCCMINOR} > 4.2 ]] ; then
+			case $(tc-arch) in
+				sparc|x86) confgcc="${confgcc} --enable-targets=all"
+				;;
+			esac
 	fi
 
+
 	tc_version_is_at_least 4.3 && set -- "$@" \
 		--with-bugurl=http://bugs.gentoo.org/ \
 		--with-pkgversion="${BRANDING_GCC_PKGVERSION}"
@@ -2505,13 +2509,7 @@ fix_libtool_libdir_paths() {
 
 is_multilib() {
 	[[ ${GCCMAJOR} < 3 ]] && return 1
-	case ${CTARGET} in
-		mips64*|powerpc64*|s390x*|sparc*|x86_64*)
-			has_multilib_profile || use multilib ;;
-		*-*-solaris*) use multilib ;;
-		*-apple-darwin*) use multilib ;;
-		*)	false ;;
-	esac
+	has_multilib_profile || use multilib
 }
 
 is_cxx() {
