Index: toolchain.eclass
===================================================================
RCS file: /var/cvsroot/gentoo-x86/eclass/toolchain.eclass,v
retrieving revision 1.510
diff -u -p -r1.510 toolchain.eclass
--- toolchain.eclass	10 Dec 2011 08:55:37 -0000	1.510
+++ toolchain.eclass	12 Dec 2011 00:41:41 -0000
@@ -979,7 +979,11 @@ gcc-compiler-configure() {
 		# Default arch for x86 is normally i386, lets give it a bump
 		# since glibc will do so based on CTARGET anyways
 		x86)
-			confgcc+=" --with-arch=${CTARGET%%-*}"
+			if ! is_multilib ; then
+				# Don't set default arch for multilib; it breaks 64bit compiles
+				# unless you give -march, which breaks the gcc build itself.
+				confgcc+=" --with-arch=${CTARGET%%-*}"
+			fi
 			;;
 		# Enable sjlj exceptions for backward compatibility on hppa
 		hppa)
