diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN babl-0.0.22.orig/configure.ac babl-0.0.22/configure.ac
--- babl-0.0.22.orig/configure.ac	2008-06-12 14:54:21.000000000 -0400
+++ babl-0.0.22/configure.ac	2009-01-16 22:46:33.180020940 -0500
@@ -159,7 +159,6 @@ case "$target_or_host" in
   x86_64-*-*)
     have_x86=yes
     AC_DEFINE(ARCH_X86, 1, [Define to 1 if you are compiling for ix86.])
-    AC_DEFINE(ARCH_X86_64, 1, [Define to 1 if you are compiling for amd64.])
     ;;
   ppc-*-* | powerpc-*)
     have_ppc=yes
