diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN avidemux_2.4.3.orig/ConfigureChecks.cmake avidemux_2.4.3/ConfigureChecks.cmake
--- avidemux_2.4.3.orig/ConfigureChecks.cmake	2008-07-23 14:58:15.000000000 -0400
+++ avidemux_2.4.3/ConfigureChecks.cmake	2009-01-18 12:16:00.256536568 -0500
@@ -29,7 +29,7 @@ ELSEIF (UNIX)
 	IF    (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "i386" OR ${CMAKE_SYSTEM_PROCESSOR} STREQUAL "i586" OR ${CMAKE_SYSTEM_PROCESSOR} STREQUAL "i686")
 		SET(ADM_CPU_X86 1)
 	ELSEIF (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64" OR ${CMAKE_SYSTEM_PROCESSOR} STREQUAL "amd64")
-		SET(ADM_CPU_X86_64 1)
+		SET(ADM_CPU_X86 1)
 	ELSEIF (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "ppc")
 		SET(ADM_CPU_PPC 1)
 	ENDIF (${CMAKE_SYSTEM_PROCESSOR} STREQUAL "i386" OR ${CMAKE_SYSTEM_PROCESSOR} STREQUAL "i586" OR ${CMAKE_SYSTEM_PROCESSOR} STREQUAL "i686")
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN avidemux_2.4.3.orig/configure.in.in avidemux_2.4.3/configure.in.in
--- avidemux_2.4.3.orig/configure.in.in	2008-07-23 14:58:15.000000000 -0400
+++ avidemux_2.4.3/configure.in.in	2009-01-18 12:15:20.421135008 -0500
@@ -1136,7 +1136,7 @@ echo "CPU type: $host_cpu"
 use_mmx=no
 
 case $host_cpu in
-	i386|i486|i586|i686)
+	i386|i486|i586|i686|x86_64|amd64)
 		have_little_endian=yes;		
 		AC_DEFINE(ARCH_X86,1,[post proc])		
 		AC_DEFINE(HAVE_MMX,1,[post proc])       
@@ -1148,20 +1148,6 @@ case $host_cpu in
 		use_mmx=yes
 		echo "X86 is ON"
 		;;
-	x86_64|amd64)
-		have_little_endian=yes;		
-		AC_DEFINE(ARCH_X86_64,1,"X86_64 AMD64 assembly")
-		AC_DEFINE(ARCH_64_BITS,1,"X86_64 AMD64 assembly")
-		AC_DEFINE(HAVE_MMX,1,[post proc])
-		AC_DEFINE(ENABLE_MMX,1,[post proc])       
-		AC_DEFINE(FPM_DEFAULT,1,[Mad ])
-		AC_DEFINE(RUNTIME_CPUDETECT,1,"Let FFmpeg decide which is faster")
-		AC_DEFINE(HAVE_FAST_UNALIGNED,1,"FFMPEG")
-		AC_DEFINE(HAVE_FAST_64BIT,1,"FFMPEG")
-		use_mmx=yes
-		have_amd64=yes                
-		echo "AMD64"
-		;;
 	 ia64)
 		have_little_endian=yes;
 		AC_DEFINE(ARCH_64_BITS,1,"X86_64 AMD64 assembly")
