diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN mozilla.orig/configure mozilla/configure
--- mozilla.orig/configure	2008-11-21 14:41:02.000000000 -0500
+++ mozilla/configure	2008-12-20 21:26:43.000000000 -0500
@@ -5443,6 +5443,7 @@ sun4u | sparc*)
     ;;
 
 x86_64 | ia64)
+    OS_TEST=i686
     CPU_ARCH="$OS_TEST"
     ;;
 esac
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN mozilla.orig/configure.in mozilla/configure.in
--- mozilla.orig/configure.in	2008-11-21 14:37:59.000000000 -0500
+++ mozilla/configure.in	2008-12-20 21:26:43.000000000 -0500
@@ -1265,6 +1265,7 @@ sun4u | sparc*)
     ;;
 
 x86_64 | ia64)
+    OS_TEST=i686
     CPU_ARCH="$OS_TEST"
     ;;
 esac
