diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN glib-2.18.3.orig/configure.in glib-2.18.3/configure.in
--- glib-2.18.3.orig/configure.in	2008-11-24 00:48:47.000000000 -0500
+++ glib-2.18.3/configure.in	2009-01-06 21:21:26.064757688 -0500
@@ -2363,8 +2363,8 @@ if test x"$GCC" = xyes; then
       ;;
     x86_64)
       AC_MSG_RESULT([x86_64])
-      AC_DEFINE_UNQUOTED(G_ATOMIC_X86_64, 1,
-		   	 [x86_64 atomic implementation])
+      AC_DEFINE_UNQUOTED(G_ATOMIC_I486, 1,
+		   	 [i486 atomic implementation])
       glib_memory_barrier_needed=no
      ;;
     powerpc*)
