diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN mkvtoolnix-2.4.2.orig/ac/ax_boost_base.m4 mkvtoolnix-2.4.2/ac/ax_boost_base.m4
--- mkvtoolnix-2.4.2.orig/ac/ax_boost_base.m4	2009-01-17 16:07:31.000000000 -0500
+++ mkvtoolnix-2.4.2/ac/ax_boost_base.m4	2009-02-23 19:14:59.000000000 -0500
@@ -87,7 +87,7 @@ AC_DEFUN([AX_BOOST_BASE],
     dnl or if you install boost with RPM
     if test "$ac_boost_path" != ""; then
         case $target in
-             ppc64*|sparc64*|x86_64*) BOOST_LDFLAGS="-L$ac_boost_path/lib64" ;;
+             ppc64*|sparc64*) BOOST_LDFLAGS="-L$ac_boost_path/lib64" ;;
              *)                       BOOST_LDFLAGS="-L$ac_boost_path/lib" ;;
         esac
         BOOST_CPPFLAGS="-I$ac_boost_path/include"
@@ -95,7 +95,7 @@ AC_DEFUN([AX_BOOST_BASE],
         for ac_boost_path_tmp in /usr /usr/local /opt /opt/local ; do
             if test -d "$ac_boost_path_tmp/include/boost" && test -r "$ac_boost_path_tmp/include/boost"; then
                 case $target in
-                     ppc64*|sparc64*|x86_64*) BOOST_LDFLAGS="-L$ac_boost_path_tmp/lib64" ;;
+                     ppc64*|sparc64*) BOOST_LDFLAGS="-L$ac_boost_path_tmp/lib64" ;;
                      *)                       BOOST_LDFLAGS="-L$ac_boost_path_tmp/lib" ;;
                 esac
                 BOOST_CPPFLAGS="-I$ac_boost_path_tmp/include"
@@ -172,7 +172,7 @@ AC_DEFUN([AX_BOOST_BASE],
             if test "$ac_boost_lib_path" = ""
             then
                 case $target in
-                     ppc64*|sparc64*|x86_64*) BOOST_LDFLAGS="-L$best_path/lib64" ;;
+                     ppc64*|sparc64*) BOOST_LDFLAGS="-L$best_path/lib64" ;;
                      *)                       BOOST_LDFLAGS="-L$best_path/lib" ;;
                 esac
             fi
diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN mkvtoolnix-2.4.2.orig/configure mkvtoolnix-2.4.2/configure
--- mkvtoolnix-2.4.2.orig/configure	2009-01-17 16:11:11.000000000 -0500
+++ mkvtoolnix-2.4.2/configure	2009-02-23 19:15:08.000000000 -0500
@@ -10095,7 +10095,7 @@ echo $ECHO_N "checking for boostlib >= $
 
                 if test "$ac_boost_path" != ""; then
         case $target in
-             ppc64*|sparc64*|x86_64*) BOOST_LDFLAGS="-L$ac_boost_path/lib64" ;;
+             ppc64*|sparc64*) BOOST_LDFLAGS="-L$ac_boost_path/lib64" ;;
              *)                       BOOST_LDFLAGS="-L$ac_boost_path/lib" ;;
         esac
         BOOST_CPPFLAGS="-I$ac_boost_path/include"
@@ -10103,7 +10103,7 @@ echo $ECHO_N "checking for boostlib >= $
         for ac_boost_path_tmp in /usr /usr/local /opt /opt/local ; do
             if test -d "$ac_boost_path_tmp/include/boost" && test -r "$ac_boost_path_tmp/include/boost"; then
                 case $target in
-                     ppc64*|sparc64*|x86_64*) BOOST_LDFLAGS="-L$ac_boost_path_tmp/lib64" ;;
+                     ppc64*|sparc64*) BOOST_LDFLAGS="-L$ac_boost_path_tmp/lib64" ;;
                      *)                       BOOST_LDFLAGS="-L$ac_boost_path_tmp/lib" ;;
                 esac
                 BOOST_CPPFLAGS="-I$ac_boost_path_tmp/include"
@@ -10227,7 +10227,7 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
             if test "$ac_boost_lib_path" = ""
             then
                 case $target in
-                     ppc64*|sparc64*|x86_64*) BOOST_LDFLAGS="-L$best_path/lib64" ;;
+                     ppc64*|sparc64*) BOOST_LDFLAGS="-L$best_path/lib64" ;;
                      *)                       BOOST_LDFLAGS="-L$best_path/lib" ;;
                 esac
             fi
