diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN gpac.orig/configure gpac/configure
--- gpac.orig/configure	2008-12-02 13:04:42.000000000 -0500
+++ gpac/configure	2009-02-24 11:43:45.000000000 -0500
@@ -222,7 +222,7 @@ case "$cpu" in
     if [ x"$canon_arch" = x"x86_64" -o x"$canon_arch" = x"amd64" ]; then
       if [ -z "`echo $CFLAGS | grep -- -m32`"  ]; then
         cpu="x86_64"
-	libdir="lib64"
+	libdir="lib32"
         #that's a bit crude...
         PIC_CFLAGS="-fPIC -DPIC"
         want_pic="yes"
@@ -1674,7 +1674,7 @@ echo "USE_X11_XV=$has_x11_xv" >> config.
 fi
 
 if test "$is_64" = "yes"; then
-echo "X11_LIB_PATH=$X11_PATH/lib64" >> config.mak
+echo "X11_LIB_PATH=$X11_PATH/lib32" >> config.mak
 else
 echo "X11_LIB_PATH=$X11_PATH/lib" >> config.mak
 fi
