diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN liboil-0.3.15.orig/m4/as-host-defines.m4 liboil-0.3.15/m4/as-host-defines.m4
--- liboil-0.3.15.orig/m4/as-host-defines.m4	2008-05-09 21:36:28.000000000 -0400
+++ liboil-0.3.15/m4/as-host-defines.m4	2008-12-19 23:10:37.000000000 -0500
@@ -16,8 +16,8 @@ AC_DEFUN([AS_HOST_DEFINES],
       AC_DEFINE(HAVE_I386, 1, [Defined if host is i386])
       ;;
     xx86_64|xamd64)
-      HAVE_AMD64=yes
-      AC_DEFINE(HAVE_AMD64, 1, [Defined if host is amd64])
+      HAVE_I386=yes
+      AC_DEFINE(HAVE_I386, 1, [Defined if host is i386])
       ;;
     xpowerpc|xppc|xpowerpc64|xppc64)
       HAVE_POWERPC=yes
